src/main/java/org/jscep/server/ScepServlet.java

```diff
@@ -111,23 +111,13 @@ public final void service(final HttpServletRequest req,
         try {
             op = getOperation(req);
             if (op == null) {
-                // The operation parameter must be set.
-
-                res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                Writer writer = res.getWriter();
-                writer.write("Missing \"operation\" parameter.");
-                writer.flush();
-
+                 // The operation parameter must be set.
+                res.sendError(HttpServletResponse.SC_BAD_REQUEST, "Missing \"operation\" parameter.");
```

Comment:
Are there any side effects from the text/html content type?

Reply:
I made a mistake about my motivation for changing this.  I somehow thought `Missing "operation" parameter.` wasn't making it to the response body because I misunderstood the test I wrote.  Let me re-organize the commits to add the test to try to make it all more clear.

When I'm using the resulting code in my spring boot app, I actually see a json response: ```{"timestamp":1529594525468,"status":400,"error":"Bad Request","message":"Missing \"operation\" parameter.","path":"/scep/"}```, with an additional `Content-Type: application/json;charset=utf-8` header that wasn't there before.

I forget what causes spring boot to respond with json instead of html.

As for side effects, I suppose it's possible depending on what the code handling the response does.  I figure it couldn't be doing much though since it's an error response, and sendError generates error responses elsewhere in this class.

Reply:
OK, I ended up force-pushing, but with any luck you can now see the initial responses, and how they change by using sendError and then adding the try-catch block.